artman/tasks/publish/local.py

```diff
@@ -53,12 +53,19 @@ def execute(self, git_repo, local_paths, output_dir,
 
         # Where is the target git repo located?
         # Start by checking for an explicit path in `local_paths`, and then
-        # if none is found, derive it from reporoot.
+        # if none is found, clone the repo to output_dir.
         repo_name_underscore = repo_name.replace('-', '_')
-        api_repo = local_paths.get(repo_name_underscore,
-            os.path.join(local_paths.get('reporoot', '..'), repo_name),
-        )
-        api_repo = os.path.realpath(os.path.expanduser(api_repo))
+        api_repo = local_paths.get(repo_name_underscore)
+        if not api_repo:
+            api_repo = os.path.join(output_dir, repo_name)
+            if os.path.exists(api_repo):
+                self.exec_command(['rm', '-rf', api_repo])
```

Comment:
Not completely sure, but does this cause a danger that someone's existing repo clone (that might contain local branches/feature work that hasn't been pushed) will get deleted and replaced with a clean checkout?

Reply:
It could in theory. We will wipe out the content only if such local setting is not specified in artman user config, and the final api_repo would be under output_dir folder. Usually, we don't expect user to mix the input and output directory.

Reply:
For example, the default output location would be ./artman-genfiles, and therefore a "api-client-staging" repo will be under ./artman-genfiles/api-client-staging.

Reply:
Ah ok, that makes sense

Reply:
> Usually, we don't expect user to mix the input and output directory.

To be clear, this is an issue only if the input is under an `artman-genfiles` directory, right?

Reply:
No. If one specify a output_dir explicitly, and that output_dir contains input files, it might still be affected.